pkg/auth/auth.go

```diff
@@ -181,37 +255,151 @@ func (c *Credentials) GetPollerAuth() (PollerAuth, error) {
 	if err != nil {
 		return PollerAuth{}, err
 	}
-	if auth.IsCert {
-		return auth, nil
-	}
-	if auth.Username != "" {
-		defaultAuth.Username = auth.Username
-	}
 	_ = mergo.Merge(&auth, defaultAuth)
 	return auth, nil
 }
 
 func getPollerAuth(c *Credentials, poller *conf.Poller) (PollerAuth, error) {
+	// by default, enforce secure TLS
+	insecureTLS := false
+	if poller.UseInsecureTLS != nil {
+		insecureTLS = *poller.UseInsecureTLS
+	}
 	if poller.AuthStyle == conf.CertificateAuth {
-		return PollerAuth{IsCert: true}, nil
+		return handCertificateAuth(c, poller, insecureTLS)
 	}
 	if poller.Password != "" {
-		return PollerAuth{Username: poller.Username, Password: poller.Password}, nil
+		return PollerAuth{Username: poller.Username, Password: poller.Password, insecureTLS: insecureTLS}, nil
 	}
 	if poller.CredentialsScript.Path != "" {
+		pass, err := c.password(poller)
+		if err != nil {
+			return PollerAuth{}, err
+		}
 		return PollerAuth{
 			Username:            poller.Username,
-			Password:            c.password(poller),
+			Password:            pass,
 			HasCredentialScript: true,
 			Schedule:            poller.CredentialsScript.Schedule,
+			insecureTLS:         insecureTLS,
 		}, nil
 	}
 	if poller.CredentialsFile != "" {
 		err := conf.ReadCredentialFile(poller.CredentialsFile, poller)
 		if err != nil {
 			return PollerAuth{}, err
 		}
-		return PollerAuth{Username: poller.Username, Password: poller.Password}, nil
+		return PollerAuth{Username: poller.Username, Password: poller.Password, insecureTLS: insecureTLS}, nil
+	}
+	return PollerAuth{Username: poller.Username, insecureTLS: insecureTLS}, nil
+}
+
+func handCertificateAuth(c *Credentials, poller *conf.Poller, insecureTLS bool) (PollerAuth, error) {
+	if poller.CertificateScript.Path != "" {
+		certBlob, err := c.certs(poller)
+		if err != nil {
+			return PollerAuth{}, err
+		}
+		cert, key, err := extractCertAndKey(certBlob)
+		if err != nil {
+			return PollerAuth{}, err
+		}
+		return PollerAuth{
+			IsCert:               true,
+			HasCertificateScript: true,
+			PemCert:              cert,
+			PemKey:               key,
+			insecureTLS:          insecureTLS,
+		}, nil
+	}
+
+	var pathPrefix string
+	certPath := poller.SslCert
+	keyPath := poller.SslKey
+
+	if certPath == "" {
+		o := &options.Options{}
+		options.SetPathsAndHostname(o)
+		pathPrefix = path.Join(o.HomePath, "cert/", o.Hostname)
```

Comment:
these 3 lines could be outside this if block as it would be used by below if case also.